### ISSUE TITLE
When local variable is used as the value to append in a loop, it does not get detected as a loop to refactor

### ISSUE THREAD

**Issue by papalotis** [sourcery-ai/sourcery#84]
### Issue description or question
Sourcery works for this function
```python
def test_func_1():
    thelist = []
    for i in range(10):
        thelist.append(i ** 2)
    return thelist
```
with the suggestion to turn it into this `return [i ** 2 for i in range(10)]`

However, when the value to be appended is stored in a local variable that is not used elsewhere sourcery is not triggered
```python
def test_func_2():
    thelist = []
    for i in range(10):
        k = i ** 2
        thelist.append(k)
    return thelist
````

I would expect sourcery to detect the variable that is used only inside the loop and remove it along with the refactor like above and if it is known that minimum python version is 3.8 maybe even store the result in the variable using `:=` (walrus operator)

### Sourcery Version

0.7.12

### Code editor or IDE name and version

VS Code 1.49.3

### OS name and version

Windows (WSL Ubuntu 16.04)

**Comment by Hellebore** [comment]
This would be neat. We will look into it.

**Comment by Hellebore** [comment]
Fixed in 0.9.1.
